cirq-core/cirq/sim/state_vector.py

```diff
@@ -314,7 +314,8 @@ def measure_state_vector(
     out.shape = initial_shape
     out /= np.sqrt(probs[result])
 
-    return measurement_bits, out
+    # We mutate and return out, so mypy cannot identify that the out cannot be None.
```

Comment:
Would it be able to do so if we added `assert out is not None` either right here or on line 307 above?

As a matter of principle, I think we should prefer code over comments whenever possible in order to help python/mypy/etc help us :-)

Reply:
Oh much better thanks!